monitor/external_export/graphana.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+import json
+import os
+import requests
+import string
+import argparse
+
+from monitor import get_lambda_names
+
+
+mon_home = os.getenv('DSS_MON_HOME')
+
+def load_template_file(template_path: str):
+    """Loads JSON template to dict."""
+    with open(template_path) as file:
+        data = json.load(file)
+    return data
+
+def format_panel_positioning(dashboard_src:dict, dashboard_dst: dict):
+    '''Use the dashboard_src to position panels correctly on the dashboard_dest'''
+    for panel in dashboard_dst['panels']:
+        for source_panel in dashboard_src.get("panels"):
+            if panel['title'] == source_panel['title']:
+                panel["gridPos"] = source_panel['gridPos']
+                break
+            print(f"Unable to locate source panel with title: {panel['title']}")
+        break
+    return dashboard_dst
+
+class DCPMetricsDash:
+    def __init__(self):
+        self.dcp_monitor_dashboard_url = 'https://raw.githubusercontent.com/HumanCellAtlas/dcp-monitoring/' \
+                                         'master/terraform/modules/env-dashboards/dss-dashboard.tf'
+        self.current_dashboard = self.get_current_dashboard()
+        self.unused_id = self.get_unused_panel_id()
+
+    def get_unused_panel_id(self):
+        ids = [ x for x in range(100) if x not in self.get_used_panel_ids() ]
+        for id in ids:
+            yield id
+
+    def get_used_panel_ids(self):
+        return [panel['id'] for panel in self.current_dashboard['panels']]
+
+    def get_current_dashboard(self):
+        """returns the terraform managed dashboard for the DSS"""
+        resp = requests.get(url=self.dcp_monitor_dashboard_url)
+        return json.loads(resp.text.split('EOF')[1])
+
+    def format_tf_templates(self, new_dashboard:dict ):
+        dashboard = json.dumps(new_dashboard,indent=2)
+        intro = 'locals {\n  dss_dashboard = <<EOF\n'
+        tail = '\nEOF\n}\n'
+        return f'{intro}{dashboard}{tail}'
+
+class DSSMetrics:
+    def __init__(self):
+        self.refid = self.get_refid()
+
+    def get_refid(self):
+        alpha_chars = [char for char in string.ascii_uppercase]
+        for char in alpha_chars:
+            yield char
+
+    def build_panel(self,filepath: str, metric_name: str, gridPos:dict, id:int, panel_title: str):
+        panel_template = load_template_file(filepath)
+        targets = self._build_targets(metric_name)
+        for target in targets:
+            panel_template['targets'].append(target)
+        panel_template["title"] = panel_title
+        panel_template["gridPos"] = gridPos
+        panel_template['id'] = id
+        return panel_template
+
+class LambdaMetrics(DSSMetrics):
+    def __init__(self):
+
+        self.lambda_names = self._get_stripped_lambda_names()
+        self.lambda_panel_template_path = mon_home+'/templates/graphana/lambda_panel_template.json'
+        self.refid = self.get_refid()
+
+    def get_refid(self):
```

Comment:
Both dss_metrics and lambda_metrics use this.  Consider making it external and again:
```
    def get_refid(self):
        for char in string.ascii_uppercase:
            yield char
```